pkg/provider/remote_state.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	be "github.com/hashicorp/terraform/backend"
+	backendInit "github.com/hashicorp/terraform/backend/init"
+	"github.com/pulumi/pulumi/sdk/proto/go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func remoteStateReferenceRead(ctx context.Context, req *pulumirpc.ReadRequest) (*pulumirpc.ReadResponse, error) {
+	// Prevent Terraform from logging minutia
+	log.SetOutput(ioutil.Discard)
```

Comment:
Curious - why do we want to do this?  Don't we want to report back logs for diagnostics puproses?

I know we have `--debug` which today gets passed through to providers and can be used to turn on verbose Terraform Provider logging - can we support this in the same way here?

Reply:
Quite possibly - as it is now, the log comes back over the standard diagnostics and feeds back INFO level messages in all cases. I'll look at how we deal with this elsewhere.

Reply:
Sorry, I mean _without_ discarding the log - not as it is now.

Reply:
See for example: https://github.com/pulumi/pulumi-terraform/blob/d089c9cf69cff4594f59d0e80c5afb7efa596fc8/pkg/tfbridge/provider.go#L196

Reply:
Yup, that's what we want.

Reply:
OK this should be resolved now.